cardinal/server/server.go

```diff
@@ -46,6 +53,152 @@ const (
+	//var txEndpoints []string
+
+	// Register tx endpoints
+	txs, err := world.ListTransactions()
+	if err != nil {
+		return err
+	}
+
+	txNameToTx := make(map[string]transaction.ITransaction)
+	for _, tx := range txs {
+		txNameToTx[tx.Name()] = tx
+	}
+
+	coreHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		return TransactionReply{
+			TxHash: "",
+			Tick:   0,
+		}, errors.New("not implemented")
+	})
+
+	createPersonaHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		return ecs.CreatePersonaTransactionResult{Success: true}, errors.New("not implemented")
+	})
+
+	authorizePersonaAddressHandler := runtime.OperationHandlerFunc(func(i interface{}) (interface{}, error) {
+		return ecs.AuthorizePersonaAddressResult{Success: true}, errors.New("not implemented")
+	})
+	api.RegisterOperation("POST", "/tx/core/{txType}", coreHandler)
+	api.RegisterOperation("POST", "/tx/persona/create-persona", createPersonaHandler)
+	api.RegisterOperation("POST", "/tx/persona/authorize-persona-address", authorizePersonaAddressHandler)
+
+	return nil
+}
+
+// result struct for /query/http/endpoints
+type EndpointsResult struct {
+	TxEndpoints    []string `json:"tx_endpoints"`
+	QueryEndpoints []string `json:"query_endpoints"`
+}
+
+// register query endpoints for swagger server
+func registerReadHandlerSwagger(world *ecs.World, api *untyped.API) error {
+	//var txEndpoints []string
+	coreHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		return struct {
+			test string
+		}{test: "test"}, errors.New("not implemented")
+	})
+	listHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		txs, err := world.ListTransactions()
+		txEndpoints := make([]string, 0, len(txs))
+		if err != nil {
+			return nil, err
+		}
+		for _, tx := range txs {
+			if tx.Name() != "create-persona" && tx.Name() != "authorize-persona-address" {
+				txEndpoints = append(txEndpoints, "/tx/core/"+tx.Name())
+			} else {
+				txEndpoints = append(txEndpoints, "/tx/persona/"+tx.Name())
+			}
+		}
+
+		queryEndpoints := make([]string, 0, len(world.ListReads())+3)
+		for _, read := range world.ListReads() {
+			queryEndpoints = append(queryEndpoints, "/query/core/"+read.Name())
+		}
+		queryEndpoints = append(queryEndpoints, "/query/http/endpoints")
+		queryEndpoints = append(queryEndpoints, "/query/persona/signer")
+		queryEndpoints = append(queryEndpoints, "/query/receipt/submit")
+		return EndpointsResult{
+			TxEndpoints:    txEndpoints,
+			QueryEndpoints: queryEndpoints,
+		}, nil
+	})
+	personaHandler := runtime.OperationHandlerFunc(func(i interface{}) (interface{}, error) {
+		return ReadPersonaSignerResponse{Status: "", SignerAddress: ""}, errors.New("not implemented")
+	})
+	receiptsHandler := runtime.OperationHandlerFunc(func(i interface{}) (interface{}, error) {
+		return struct {
+			start_tick uint64
```

Comment:
What's the purpose of returning these empty structs? Will these function eventually return actual structs, and this code is just serving as samples for what it will eventually look like?

Reply:
Yeah, it's just stubs. The anonymous structs are patterned after the swagger spec file.